### PR TITLE
Create the SmartAds package

### LINKED ISSUE #400: Head and settings of SmartAds package [5pt]
**Feature:** [Smart Adserver](https://community.frontity.org/t/smart-adserver/1586).
**User Story**: Configure Smart Adserver service installing a Frontity package.

[**Sprint Card**](https://github.com/orgs/frontity/projects/4#card-37072939)

### LINKED ISSUE #401: SmartAd component [5pt]
**Feature:** [Smart Adserver](https://community.frontity.org/t/smart-adserver/1586).
**User Stories**: 
  - Create fills (see [Slot and Fill](https://community.frontity.org/t/slot-and-fill/895))with ad components.
  - Use the Ad component directly.

[**Sprint Card**](https://github.com/orgs/frontity/projects/4#card-37072939)

### PR DESCRIPTION
**Pull request by michalczaplinski**
**What**:

Create the new `@frontity/smart-adserver` package. 

Following the Feature Discussion: https://community.frontity.org/t/smart-adserver/1586

**Why**:

To integrate Smart Adserver into Frontity.

**How**:

- Create the new `@frontity/smart-adserver` package in the frontity-monorepo
- Expose the root which will include the `<Head />` that loads the Smart Adserver library
- Allow the user to pass settings to the package using `frontity.settings.js`
- Create the `SmartAd` component that will be used in the `libraries`.

**Tasks**:

- [x] Code
- [x] TSDocs
- [x] TypeScript
- [x] Unit tests
- [x] End to end tests
- [x] Open an issue for this feature in the [docs repo](https://github.com/frontity/docs/wiki/Code-Releases): https://github.com/frontity/docs/issues/234
- [x] Update community discussions: https://community.frontity.org/t/smart-adserver/1586/10
- [x] Add a changeset (with link to its [Feature Discussion](https://community.frontity.org/c/33) if it exists)

**Unrelated Tasks**

- [x] TypeScript tests
- [x] Update starter themes
- [x] Update other packages

### CODE REVIEW

**Comment by DAreRodz** [comment]
> **Additional Comments**
>
> :warning: The `SmartAd` component will be implemented in another PR.

Are you sure? It looks like you [already implemented it](https://github.com/frontity/frontity/pull/569/files#diff-6d9700a19725c6d1912655373318ef4c). :joy:

**Comment by michalczaplinski** [comment]
> > **Additional Comments**
> > ⚠️ The `SmartAd` component will be implemented in another PR.
> 
> Are you sure? It looks like you [already implemented it](https://github.com/frontity/frontity/pull/569/files#diff-6d9700a19725c6d1912655373318ef4c). 😂

I think it'll make more sense to implement them together so that I can just test them with the e2e tests 🙂

**Comment by michalczaplinski** [comment]
Screen Shot 2020-09-29 at 9 46 26 AM

new laptop, I've realized that I didn't set my git email, whoops

File: .eslintrc [hunk 1]
```diff
@@ -8,8 +8,10 @@
     "plugin:jest/recommended",
     "plugin:cypress/recommended"
   ],
-  "plugins": ["jsdoc"],
+  "plugins": ["jsdoc", "react-hooks"],
```
Comment (michalczaplinski) [inline .eslintrc:11]: This is the official eslint plugin from the react team: https://www.npmjs.com/package/eslint-plugin-react-hooks

File: packages/smart-adserver/src/components/smart-ad.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * @returns The styles.
+ */
+const styles = (callType: string, width: number, height: number) =>
+  callType === "iframe"
+    ? css`
+        width: ${width}px;
+        height: ${height}px;
+      `
+    : css``;
```
Comment (luisherranz) [inline packages/smart-adserver/src/components/smart-ad.tsx]: Can't we also style the div of `std`?

Ads are always a mess for Cumulative Layout Shift (https://web.dev/cls/) so I remember that in the past we always tried to add a `height`.

Maybe the `height` value can be `height` for the iframe and `min-height` for the `div`, so if a bigger ad is loaded it's not cropped.

@darerodz, what do you think?

**Review by luisherranz — COMMENTED** [review]
Great work Michal!

I didn't approve it because I'd prefer @darerodz to do so. He has more experience with ads than me, and he worked on our old smartads package and the current `@frontity/google-ad-manager` package.

Other than that, I've simplified the E2E tests and switched to a `major` bump for the 1.0.0 release.

I saw there are some Eslint warnings left too. I think they are coming from the new `react-hooks` eslint package.

File: packages/smart-adserver/src/components/smart-ad.tsx (hunk 2 above)
Comment (DAreRodz) [reply packages/smart-adserver/src/components/smart-ad.tsx]: I agree. :+1:
I think we did something similar in the past.

**Comment by michalczaplinski** [comment]
@DAreRodz You're welcome to review this. Some notes for you:

1. I've allowed users to pass a `css` prop to add additional styles to the ad container as well as pass any other props to the container.

2. I've actually added all the dependencies suggested by the react-hooks eslint plugin because I've realized that it shouldn't be any problem. So, no eslint warnings anymore.

3. The `minHeight` prop on the `SmartAd` component on the does not have any default value, so the user has to pass it themselves, otherwise no height is specified.

**Review by DAreRodz — APPROVED** [review]
Everything all right. Great job @michalczaplinski!

PS: I just fixed two minor typos.

**Pull request merged**